src/Utils/FindBreakingChanges.php

```diff
@@ -0,0 +1,583 @@
+    const BREAKING_CHANGE_TYPE_CHANGED = 'TYPE_CHANGED_KIND';
+    const BREAKING_CHANGE_TYPE_REMOVED = 'TYPE_REMOVED';
+    const BREAKING_CHANGE_TYPE_REMOVED_FROM_UNION = 'TYPE_REMOVED_FROM_UNION';
+    const BREAKING_CHANGE_VALUE_REMOVED_FROM_ENUM = 'VALUE_REMOVED_FROM_ENUM';
+    const BREAKING_CHANGE_ARG_REMOVED = 'ARG_REMOVED';
+    const BREAKING_CHANGE_ARG_CHANGED = 'ARG_CHANGED_KIND';
+    const BREAKING_CHANGE_NON_NULL_ARG_ADDED = 'NON_NULL_ARG_ADDED';
+    const BREAKING_CHANGE_NON_NULL_INPUT_FIELD_ADDED = 'NON_NULL_INPUT_FIELD_ADDED';
+    const BREAKING_CHANGE_INTERFACE_REMOVED_FROM_OBJECT = 'INTERFACE_REMOVED_FROM_OBJECT';
+
+    const DANGEROUS_CHANGE_ARG_DEFAULT_VALUE = 'ARG_DEFAULT_VALUE_CHANGE';
+    const DANGEROUS_CHANGE_VALUE_ADDED_TO_ENUM = 'VALUE_ADDED_TO_ENUM';
+    const DANGEROUS_CHANGE_TYPE_ADDED_TO_UNION = 'TYPE_ADDED_TO_UNION';
+
+    /**
+     * Given two schemas, returns an Array containing descriptions of all the types
+     * of potentially dangerous changes covered by the other functions down below.
+     *
+     * @return array
+     */
+    public static function findDangerousChanges(Schema $oldSchema, Schema $newSchema)
+    {
+        return array_merge(self::findArgChanges($oldSchema, $newSchema)['dangerousChanges'],
+            self::findValuesAddedToEnums($oldSchema, $newSchema),
+            self::findTypesAddedToUnions($oldSchema, $newSchema)
+        );
+    }
+
+    /**
+     * Given two schemas, returns an Array containing descriptions of all the types
+     * of breaking changes covered by the other functions down below.
+     *
+     * @return array
+     */
+    public static function findBreakingChanges(Schema $oldSchema, Schema $newSchema)
+    {
+        return array_merge(
+            self::findRemovedTypes($oldSchema, $newSchema),
+            self::findTypesThatChangedKind($oldSchema, $newSchema),
+            self::findFieldsThatChangedType($oldSchema, $newSchema),
+            self::findTypesRemovedFromUnions($oldSchema, $newSchema),
+            self::findValuesRemovedFromEnums($oldSchema, $newSchema),
+            self::findArgChanges($oldSchema, $newSchema)['breakingChanges'],
+            self::findInterfacesRemovedFromObjectTypes($oldSchema, $newSchema)
+        );
+    }
+
+    /**
+     * Given two schemas, returns an Array containing descriptions of any breaking
+     * changes in the newSchema related to removing an entire type.
+     *
+     * @return array
+     */
+    public static function findRemovedTypes(
+        Schema $oldSchema, Schema $newSchema
+    )
+    {
+        $oldTypeMap = $oldSchema->getTypeMap();
+        $newTypeMap = $newSchema->getTypeMap();
+
+        $breakingChanges = [];
+        foreach ($oldTypeMap as $typeName => $typeDefinition) {
+            if (!isset($newTypeMap[$typeName])) {
+                $breakingChanges[] =
+                    ['type' => self::BREAKING_CHANGE_TYPE_REMOVED, 'description' => "${typeName} was removed."];
```

Comment:
Had no idea that `${typeName}` is a valid form of variable interpolation in PHP. Cool %)